plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -132,6 +133,40 @@ export class HogTransformerService {
                 // For now, execute each transformation function in sequence
                 for (const hogFunction of teamHogFunctions) {
                     const transformationIdentifier = `${hogFunction.name} (${hogFunction.id})`
+
+                    // Check if we should apply this transformation based on its filters
+                    if (this.hub.FILTER_TRANSFORMATIONS_ENABLED) {
+                        const globals = this.createInvocationGlobals(event)
+                        const filterGlobals = convertToHogFunctionFilterGlobal(globals)
+                        const shouldApplyTransformation = this.checkEventAgainstFilters(hogFunction, filterGlobals)
+
+                        if (!shouldApplyTransformation) {
+                            logger.info('🔍', 'Skipping transformation due to filter mismatch', {
+                                event_id: event.uuid,
+                                transformation_id: hogFunction.id,
+                                transformation_name: hogFunction.name,
+                            })
+                            // Add a log entry to the results when filtered out
+                            const globalsWithInputs = buildGlobalsWithInputs(globals, {
+                                ...(hogFunction.inputs ?? {}),
+                                ...(hogFunction.encrypted_inputs ?? {}),
+                            })
+
+                            results.push({
+                                invocation: createInvocation(globalsWithInputs, hogFunction),
+                                logs: [
+                                    {
+                                        level: 'info',
+                                        timestamp: DateTime.now(),
+                                        message: `Transformation was skipped - filter did not match`,
+                                    },
+                                ],
+                                finished: true,
+                            })
```

Comment:
_also_ I don't think we should log this. Its just a huge amount of noise. We can just track an appmetric instead which is what we do in the destinatons side.